treeherder/log_parser/failureline.py

```diff
@@ -94,13 +96,22 @@ def get_kwargs(failure_line):
             if key in failure_line}
 
 
+def create_failure_line(job_log, failure_line):
+    fl = FailureLine.objects.create(repository=job_log.job.repository,
+                                    job_guid=job_log.job.guid,
+                                    job_log=job_log,
+                                    **get_kwargs(failure_line))
+    if "group" in failure_line:
+        # Store as unix path separators and omit the filename
+        group, saved = Group.objects.get_or_create(
+            name=path.dirname(path.join(*failure_line["group"].split("\\"))))
```

Comment:
Argh, forgot to save my updated comment:

This is really `"/".join(failure_line["group"].split("\\")).rsplit("/", 1)[0]`, at least in the case where the path isn't `/`, right? It doesn't seem worth going through the path mechanics in the case that we are explicitly trying to get one possible output, but if we must then we should at least use `posixpath` explicitly.